Runtime/Experimental/CinemachineNewVirtualCamera.cs

```diff
@@ -255,10 +265,10 @@ protected void UpdateComponentCache()
                 }
             }
 #endif
-            if (m_Components != null && m_Components.Length == (int)CinemachineCore.Stage.Finalize)
+            if (m_Components != null && m_Components.Length == (int)CinemachineCore.Stage.Finalize + 1)
```

Comment:
Why not add BodyAfterAim to CinemachineCore.Stage instead of adding a 1 to Finalize?


```
public enum Stage
{
    /// <summary>Second stage: position the camera in space</summary>
    Body,

    /// <summary>Third stage: orient the camera to point at the target</summary>
    Aim,

    /// <summary>Optional Fourth stage: for procedural placement that depends on aim</summary>
    BodyAfterAim, 

    /// <summary>Final pipeline stage: apply noise (this is done separately, in the
    /// Correction channel of the CameraState)</summary>
    Noise,

    /// <summary>Not a pipeline stage.  This is invoked on all virtual camera
    /// types, after the pipeline is complete</summary>
    Finalize
};
```

A problem with this solution, because it will give empty loop iterations in a lot of places, e.g.:
```
for (CinemachineCore.Stage stage = CinemachineCore.Stage.Body;stage <= CinemachineCore.Stage.Finalize; ++stage)
{ ... }
```